samtranslator/policy_template_processor/template.py

```diff
@@ -50,17 +53,53 @@ def to_statement(self, parameter_values):  # type: ignore[no-untyped-def]
         # injection of values for parameters not intended in the template. This is important because "Ref" resolution
         # will substitute any references for which a value is provided.
         necessary_parameter_values = {
-            name: value for name, value in parameter_values.items() if name in self.parameters
+            POLICY_PARAMETER_DUSAMBIGUATE_PREFIX + name: value
+            for name, value in parameter_values.items()
+            if name in self.parameters
         }
 
         # Only "Ref" is supported
         supported_intrinsics = {RefAction.intrinsic_name: RefAction()}
 
         resolver = IntrinsicsResolver(necessary_parameter_values, supported_intrinsics)  # type: ignore[no-untyped-call]
-        definition_copy = copy.deepcopy(self.definition)
+        definition_copy = self._disambiguate_policy_parameter(self.definition)
 
         return resolver.resolve_parameter_refs(definition_copy)
 
+    @staticmethod
+    def _disambiguate_policy_parameter(policy_definition: Any) -> Any:
+        """
+        Return a deepcopy of policy definition where all parameters are
+        renamed to avoid naming collusion of normal CFN parameters.
```

Comment:
collusion -> collision

Reply:
damn! thank you Andrew!

Reply:
fixed in a commit named "[Fix a type](https://github.com/aws/serverless-application-model/pull/2751/commits/6a4dd081f3e01d1351e8ea4421f131fc1d269d33)"